core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala

```diff
@@ -222,4 +222,56 @@ class SparkThrowableSuite extends SparkFunSuite {
         assert(false)
     }
   }
+
+  test("get message in the specified format") {
+    import ErrorMessageFormat._
+    class TestQueryContext extends QueryContext {
+      override val objectName = "v1"
+      override val objectType = "VIEW"
+      override val startIndex = 2
+      override val stopIndex = -1
+      override val fragment = "1 / 0"
+    }
+    val e = new SparkArithmeticException(
+      errorClass = "DIVIDE_BY_ZERO",
+      errorSubClass = None,
+      messageParameters = Array("CONFIG"),
+      context = Array(new TestQueryContext),
+      summary = "Query summary")
+
+    assert(SparkThrowableHelper.getMessage(e, PRETTY) ===
+      "[DIVIDE_BY_ZERO] Division by zero. Use `try_divide` to tolerate divisor being 0 " +
+      "and return NULL instead. If necessary set CONFIG to \"false\" to bypass this error." +
+      "\nQuery summary")
+    assert(SparkThrowableHelper.getMessage(e, MINIMAL) ===
+      """{"errorClass":"DIVIDE_BY_ZERO","sqlState":"22012",""" +
+      """"messageParameters":{"config":"CONFIG"},"queryContext":[{"objectType":"VIEW",""" +
+      """"objectName":"v1","startIndex":3,"fragment":"1 / 0"}]}""")
```

Comment:
The only benefit of using compact format is to save space, while I think this doesn't matter for error messages. Shall we always make it human readable?

Reply:
ok. I will print MINIMAL in the pretty form.